src/libstore/build.cc

```diff
@@ -4706,27 +4711,36 @@ void Worker::childTerminated(Goal * goal, bool wakeSleepers)
     }
 }
 
+void Worker::updateBlockingResourceName(string resourceName) {
+    if (blockingResourceName != resourceName) {
+        blockingResourceName = resourceName;
+        printError(format("waiting for %1%...") % resourceName);
+    }
+}
 
-void Worker::waitForBuildSlot(GoalPtr goal)
+void Worker::waitForBuildSlot(GoalPtr goal, string resourceName)
 {
-    debug("wait for build slot");
+    debug("wait for build slots");
+    updateBlockingResourceName(resourceName);
```

Comment:
I think this line will print the message even if we are not actually waiting, i.e. if we take the first branch in the `if` below. I’m not sure if that is the desired behaviour or not.

Reply:
Yes, thanks for noticing that. I think the reason I did that is that `debug("wait for build slot")` was here...